### DISCUSSION TITLE
NengoDL 3.0 pre-release

### DISCUSSION THREAD

**Discussion by cnrgbot** [nengo/nengo-zoo#1393] [Announcements]
> _Archived from the Nengo Discourse forum._ Originally posted by **drasmuss** on 2019-10-31. (Forum topic #993, post #1 / id 2623)

The NengoDL team has been working on a large rewrite of the NengoDL framework to take full advantage of the new features in TensorFlow 2.0.  We're excited to announce that that update is now available for anyone that would like to try out an early preview!

To try it out you will need to do a [developer installation](https://www.nengo.ai/nengo-dl/installation.html#developer-installation):

``` bash
git clone https://github.com/nengo/nengo-dl.git
pip install -e ./nengo-dl
```
or
``` bash
pip install git+https://github.com/nengo/nengo-dl.git
```

The most significant change is that we have reworked the `Simulator` API to align more closely with the Keras `Model` API.  So if you are familiar with the Keras API (e.g. `fit/compile/evaluate`), then the new Simulator API should look familiar.

We have also made some changes to the way `TensorNodes` work, and we'd recommend using the new `nengo_dl.Layer` instead of `nengo_dl.tensor_layer`.  Again, `nengo_dl.Layer` should look familiar to anyone who has used the Keras functional Layer API!

There are many other changes and improvements, check out [the changelog](https://www.nengo.ai/nengo-dl/project.html#release-history) for all the details.  The documentation and examples at https://www.nengo.ai/nengo-dl have all been updated for 3.0 as well.

We'll be adding a migration guide soon to help with the transition to NengoDL 3.0. But if you have any questions, or find something that isn't working properly, please reply to this thread and we'll get it sorted out!
